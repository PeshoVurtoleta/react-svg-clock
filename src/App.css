:root {
    background: #FFC573;
}

body {
    margin: 0;
    height: 100vh;
    display: flex;
    justify-content: center;
    align-items: center;
}

svg {
    height: 100vh;
    stroke: #0047AB;
    /*transform: rotate(-90deg);*/
}

circle {
    fill: antiquewhite;
}