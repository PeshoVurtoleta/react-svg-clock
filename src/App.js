import React from 'react';

import './App.css';

import {ClockMarks, SecondsArrow, MinutesArrow, HoursArrow, Clock, Circle} from "./components";

function App() {
    return <div className="App">
        <Clock render={(secondsDeg, minutesDeg, hoursDeg) => {
            return <>
                <Circle x="20" y="20" r="19"/>

                <ClockMarks/>
                <Circle x="25" y="25" r="4" className="Seconds__circle"/>
                <SecondsArrow deg={secondsDeg}/>
                <Circle x="25" y="25" r="0.3" className="Seconds__circle"/>
                <HoursArrow deg={hoursDeg}/>
                <MinutesArrow deg={minutesDeg}/>
                <Circle x="20" y="20" r="0.6" className="Clock__pin"/>
            </>
        }}/>
    </div>;
}

// points="14,25 18,25"

export default App;
