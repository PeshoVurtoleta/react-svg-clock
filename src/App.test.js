import React from 'react';
import { shallow, render } from "enzyme";
import App from './App';

test('<App />', () => {
  const wrapper = render(<App />);

  expect(wrapper).toMatchSnapshot();
});