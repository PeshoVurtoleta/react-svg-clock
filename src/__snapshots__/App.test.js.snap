// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> 1`] = `
<div
  class="App"
>
  <svg
    viewBox="0 0 40 40"
  >
    <circle
      class=""
      cx="20"
      cy="20"
      r="19"
    />
    <g
      class="Clock__marks"
    >
      <line
        class="Clock__mark"
        transform="rotate(0 20 20)"
        x1="20"
        x2="20"
        y1="3"
        y2="5"
      />
      <line
        class="Clock__mark"
        transform="rotate(30 20 20)"
        x1="20"
        x2="20"
        y1="3"
        y2="5"
      />
      <line
        class="Clock__mark"
        transform="rotate(60 20 20)"
        x1="20"
        x2="20"
        y1="3"
        y2="5"
      />
      <line
        class="Clock__mark"
        transform="rotate(90 20 20)"
        x1="20"
        x2="20"
        y1="3"
        y2="5"
      />
      <line
        class="Clock__mark"
        transform="rotate(120 20 20)"
        x1="20"
        x2="20"
        y1="3"
        y2="5"
      />
      <line
        class="Clock__mark"
        transform="rotate(150 20 20)"
        x1="20"
        x2="20"
        y1="3"
        y2="5"
      />
      <line
        class="Clock__mark"
        transform="rotate(180 20 20)"
        x1="20"
        x2="20"
        y1="3"
        y2="5"
      />
      <line
        class="Clock__mark"
        transform="rotate(210 20 20)"
        x1="20"
        x2="20"
        y1="3"
        y2="5"
      />
      <line
        class="Clock__mark"
        transform="rotate(240 20 20)"
        x1="20"
        x2="20"
        y1="3"
        y2="5"
      />
      <line
        class="Clock__mark"
        transform="rotate(270 20 20)"
        x1="20"
        x2="20"
        y1="3"
        y2="5"
      />
      <line
        class="Clock__mark"
        transform="rotate(300 20 20)"
        x1="20"
        x2="20"
        y1="3"
        y2="5"
      />
      <line
        class="Clock__mark"
        transform="rotate(330 20 20)"
        x1="20"
        x2="20"
        y1="3"
        y2="5"
      />
    </g>
    <circle
      class="Seconds__circle"
      cx="25"
      cy="25"
      r="4"
    />
    <polygon
      class="Clock__seconds"
      points="25,26 25,22"
    >
      <animateTransform
        attributeName="transform"
        begin="0s"
        dur="60s"
        from="0 25 25"
        repeatCount="indefinite"
        to="360 25 25"
        type="rotate"
      />
    </polygon>
    <circle
      class="Seconds__circle"
      cx="25"
      cy="25"
      r="0.3"
    />
    <polygon
      class="Clock__hours"
      points="20,21.5 20,10"
    >
      <animateTransform
        attributeName="transform"
        begin="0s"
        dur="43200s"
        from="0 20 20"
        repeatCount="indefinite"
        to="360 20 20"
        type="rotate"
      />
    </polygon>
    <polygon
      class="Clock__minutes"
      points="20,22 20,6"
    >
      <animateTransform
        attributeName="transform"
        begin="0s"
        dur="3600s"
        from="0 20 20"
        repeatCount="indefinite"
        to="360 20 20"
        type="rotate"
      />
    </polygon>
    <circle
      class="Clock__pin"
      cx="20"
      cy="20"
      r="0.6"
    />
  </svg>
</div>
`;
